File: skillful_network_server/src/main/java/fr/uca/cdr/skillful_network/entities/application/Application.java
package fr.uca.cdr.skillful_network.entities.application;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import fr.uca.cdr.skillful_network.entities.user.User;

import javax.persistence.*;
import java.util.Date;
import java.util.Objects;

@MappedSuperclass
public abstract class Application {

    public enum ApplicationStatus {INIT, SUBMITTED, INVESTIGATING, WAITING, PAUSED, POSTPONED, ACCEPTED, REJECTED}

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    protected long id;

    @ManyToOne(cascade = CascadeType.REFRESH)
    @JsonManagedReference
    protected User user;

    protected ApplicationStatus status;

    protected Date submitDate;

    public Application() {

    }

    public Application(User user, ApplicationStatus status, Date submitDate) {
        this.user = user;
        this.status = status;
        this.submitDate = submitDate;
    }

    public long getId() {
        return id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public ApplicationStatus getStatus() {
        return status;
    }

    public void setStatus(ApplicationStatus status) {
        this.status = status;
    }

    public Date getSubmitDate() {
        return submitDate;
    }

    public void setSubmitDate(Date submitDate) {
        this.submitDate = submitDate;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Application that = (Application) o;
        return id == that.id;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: skillful_network_client/src/app/shared/models/application/application.ts
import { User } from '../user/user';

export class Application {

    private _id: number;
    private _user: User;
    private _submitDate: Date;
    private _status: String;

    constructor(data: any) {
        this.id = data.id;
        this.status = data.status;
        this.user = data.user;
        this.submitDate = data.submitDate;
    }

    public get submitDate(): Date {
        return this._submitDate;
    }

    public set submitDate(submitDate: Date) {
        this._submitDate = submitDate;
    }

    public get user(): User {
        return this._user;
    }

    public set user(user: User) {
        this._user = user;
    }

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get status(): String {
        return this._status;
    }
    
    public set status(value: String) {
        this._status = value;
    }

}
